File: java/patching-array.java
class Solution {
    public int minPatches(int[] nums, int n) {
        ArrayList<Long> appliedPatches = new ArrayList<>();
        long patch = 1;
        int i = 0;

        while (patch <= n) {
            if (i < nums.length && nums[i] <= patch) {
                patch += nums[i];
                i++;
            } else {
                appliedPatches.add(patch);
                patch += patch;
            }
        }

        return appliedPatches.size();
    }
}

File: golang/PatchingArray.go
func minPatches(nums []int, n int) int {
    patches := 0
    length := len(nums)
    index := 0
    patch:= 1
    for patch <= n{
        if index < length && nums[index] <= patch{
            patch += nums[index]
            index++
        }else {
            patch += patch
            patches++
        }

    }
    return patches
}

